mailpoet/lib/Automation/Engine/Storage/AutomationRunStorage.php

```diff
@@ -108,6 +109,26 @@ function(array $subjectData) use ($runData): bool {
     );
   }
 
+  /**
+   * @param Automation $automation
+   * @return int
+   */
+  public function countRunsForAutomationAndSubject(Automation $automation, Subject $subject): int {
+    $table = esc_sql($this->table);
+    $subjectTable = esc_sql($this->subjectTable);
+
+    $sql = "SELECT count(runs.id) as count from $table as runs
```

Comment:
Might be safer to use `COUNT (DISTINCT ...)` in case the JOIN ever multiplies rows.

Reply:
b2bcb234b